API/Features/Image/ImageController.cs

```diff
@@ -1,63 +1,342 @@
+        var urlRoots = getUrlRoots();
+        var collection = new HydraCollection<DLCS.HydraModel.Image>
         {
-            this.mediator = mediator;
+            WithContext = true,
+            Members = spaceImagesResult.PageOfAssets.Assets
+                .Select(a => a.ToHydra(urlRoots)).ToArray(),
+            TotalItems = spaceImagesResult.PageOfAssets.Total,
+            PageSize = pageSize,
+            Id = Request.GetJsonLdId()
+        };
+        PartialCollectionView.AddPaging(collection, page.Value, pageSize.Value, orderByField, descending);
+        return Ok(collection);
+    }
+    
+    
+    // ############################# PUT IMAGE ###############################
+    // ############################# PATCH IMAGE #############################
+    /// <summary>
+    /// PUT an image to its ID location.
+    /// This may be a create or an update operation.
+    /// Unlike Deliverator, the same method handles a PATCH.
+    /// </summary>
+    /// <param name="customerId">(from resource path)</param>
+    /// <param name="spaceId">(from resource path)</param>
+    /// <param name="imageId">(from resource path)</param>
+    /// <param name="hydraAsset">The body of the request contains the Asset in Hydra JSON-LD form (Image class)</param>
+    /// <returns>The created or updated Hydra Image object for the Asset</returns>
+    [ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(DLCS.HydraModel.Image))] // for Patch
+    [ProducesResponseType((int)HttpStatusCode.Created, Type = typeof(DLCS.HydraModel.Image))] // for PUT when created
+    [ProducesResponseType((int)HttpStatusCode.BadRequest, Type = typeof(ProblemDetails))]
+    [ProducesResponseType((int)HttpStatusCode.MethodNotAllowed, Type = typeof(ProblemDetails))]
+    [ProducesResponseType((int)HttpStatusCode.NotFound, Type = typeof(ProblemDetails))]
+    [ProducesResponseType((int)HttpStatusCode.InsufficientStorage, Type = typeof(ProblemDetails))]
+    [ProducesResponseType((int)HttpStatusCode.NotImplemented, Type = typeof(ProblemDetails))]
+    [ProducesResponseType((int)HttpStatusCode.InternalServerError, Type = typeof(ProblemDetails))]
+    [HttpPut]
+    [HttpPatch]
+    [Route("{imageId}")]
+    public async Task<IActionResult> Image(
+        [FromRoute] int customerId,
+        [FromRoute] int spaceId,
+        [FromRoute] string imageId,
+        [FromBody] DLCS.HydraModel.Image hydraAsset)
+    {
+        // DELIVERATOR: https://github.com/digirati-co-uk/deliverator/blob/master/API/Architecture/Request/API/Entities/CustomerSpaceImage.cs#L74
+            
+        var assetId = new AssetId(customerId, spaceId, imageId);
+        var asset = hydraAsset.ToDlcsModel(customerId, spaceId, imageId);
+        asset.Id = assetId.ToString();
+    
+        // In the special case where we were passed ImageWithFile from the IngestBytes action, 
+        // it was a POST - but we should revisit that as the direct image ingest should be a PUT as well I think

```

Comment:
Is it worth creating tickets for stuff like this to come back to? Easier to track when done/not done than a comment.